app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    saved_results = {}
+
+    def inner(*args) -> Any:
+        if args in saved_results:
+            print("Getting from cache")
+            return saved_results[args]
+
+        print("Calculating new result")
+        saved_results[args] = func(*args)
+        return saved_results[args]
```

Comment:
u are using return for same thing twice, could u make this, with only one return ? )